app/assets/javascripts/initializers/initializeDrawerSliders.js

```diff
@@ -1,57 +1,51 @@
+// app/assets/javascripts/initializers
+const drawerSliders = [
+  { selector: 'sidebar-bg-left', swipeState: 'middle', side: 'left', view: 'outOfView' },
+  { selector: 'sidebar-bg-right', swipeState: 'middle', side: 'right', view: 'outOfView' },
+  { selector: 'on-page-nav-butt-left', swipeState: 'left', side: 'left', view: 'intoView' },
+  { selector: 'on-page-nav-butt-right', swipeState: 'right', side: 'right', view: 'intoView' },
+];
+
 function initializeDrawerSliders() {
-  if(!initializeSwipeGestures.called) {
-    swipeState = "middle";
+  if (!initializeSwipeGestures.called) {
+    swipeState = 'middle';
     initializeSwipeGestures();
   }
-  if (document.getElementById("on-page-nav-controls")){
-    if (document.getElementById("sidebar-bg-left")){
-      document.getElementById("sidebar-bg-left").onclick = function(){
-        swipeState = "middle";
-        slideSidebar("left","outOfView");
-      }
-    }
-    if (document.getElementById("sidebar-bg-right")){
-      document.getElementById("sidebar-bg-right").onclick = function(){
-        swipeState = "middle";
-        slideSidebar("right","outOfView");
-      }
-    }
 
-    if (document.getElementById("on-page-nav-butt-left")){
-      document.getElementById("on-page-nav-butt-left").onclick = function(){
-        swipeState = "left"
-        slideSidebar("left","intoView");
+  if (document.getElementById('on-page-nav-controls')) {
+    drawerSliders.forEach(drawerSlider => {
+      const element = document.getElementById(drawerSlider.selector);
+      if (element) {
+        element.onclick = function() {
+          swipeState = drawerSlider.swipeState;
+          slideSidebar(drawerSlider.side, drawerSlider.view);
+        };
       }
-    }
-    if (document.getElementById("on-page-nav-butt-right")){
-      document.getElementById("on-page-nav-butt-right").onclick = function(){
-        swipeState = "right"
-        slideSidebar("right","intoView");
-      }
-    }
+    });
+
     InstantClick.on('change', function() {
-      document.getElementsByTagName('body')[0].classList.remove('modal-open');
-      slideSidebar("right","outOfView");
-      slideSidebar("left","outOfView");
+      document.body.classList.remove('modal-open');
+      slideSidebar('right', 'outOfView');
+      slideSidebar('left', 'outOfView');
     });
     listenForNarrowMenuClick();
   }
 }
 
 function listenForNarrowMenuClick(event) {
-  var navLinks = document.getElementsByClassName("narrow-nav-menu");
-  var narrowFeedButt = document.getElementById("narrow-feed-butt");
-  for (var i = 0; i < navLinks.length; i++) {
-    document.getElementById("narrow-nav-menu").classList.remove("showing");
+  const navLinks = document.getElementsByClassName('narrow-nav-menu');
+  const narrowFeedButt = document.getElementById('narrow-feed-butt');
+  for (let x = 0; x < navLinks.length; x++) {
+    document.getElementById('narrow-nav-menu').classList.remove('showing');
   }
```

Comment:
this could also be a forEach:
```js
navLinks.forEeach(() => {
  document.getElementById('narrow-nav-menu').classList.remove('showing');
})
```

Reply:
also this: `() => {
  document.getElementById('narrow-nav-menu').classList.remove('showing');
}`
could be an apart function:
```js
const hideNavMenu = () => document.getElementById('narrow-nav-menu').classList.remove('showing')

navLinks.forEach(hideNavMenu)

//and then down there
navLinks.forEach(navLink => {
  navLink.addEventListener('click', hideNavMenu)
})
```